modules/steam.nix

```diff
@@ -247,6 +247,35 @@ in
           '';
         };
 
+        autoStart = mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+            Whether to automatically launch the Steam Deck UI on boot.
+
+            No Display Managers may be enabled in conjunction with this option.
+          '';
+        };
+
+        user = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+            The user to run Steam with.
+
+            This user must be in the `networkmanager` group to be able to
+            complete the OOBE (Out-of-Box Experience).
+          '';
+        };
+
+        desktopSession = mkOption {
+          type = types.str;
+          example = "plasma";
+          description = ''
+            The session to launch for Desktop Mode.
+          '';
+        };
```

Comment:
Can we make it so it works without setting a `desktopSession`?

I'm thinking about someone who wants to setup a strictly "for games" system, especially maintained for someone else.

Setting a default to `"steam-session"` should handle that use-case. Though we might want to add a warning on that, in case users end-up not changing it and expect another session?

Reply:
Might be worth document what's the proper way to find out the value to input for desktopSession here.

A bit of search telling me nixos's '/usr/share' = '/run/current-system/sw/share', but I cannot find either `wayland-sessions` or `xsessions` folder under '/run/current-system/sw/share', maybe i'm going the wrong direction here..

Reply:
Right, we should document this. There is a `-desktops` path in `$XDG_DATA_DIRS` which contains all sessions.